### ISSUE TITLE
JSX Parsing Error with Emoji in Attribute Value

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#1769]
- Markuplint version: `4.x.x`
- Parser lang: JSX
- Node.js version: `22.x.x`
- OS: macOS

## Describe the bug

When using an emoji within an attribute value in JSX, a parsing error occurs if the value is enclosed in curly braces. The issue does not occur if the emoji is enclosed in double quotes.

## Code Example or Playground URL

```jsx
<MyComponent attr={'🐱'} /> <!-- 🙀 ERROR -->
```

The issue does not occur when the emoji is enclosed in double quotes.
 
```jsx
<MyComponent attr="🐾" /> <!-- 😻 OK -->
```

https://playground.markuplint.dev/#N4Igxg9gdgZglgcxALhMAOlABF9ICmAHgC75QAmAznslgNqY454C2AhgE4DWArgA4AbOFGLIO+SCxZly+cnkZYAugBpFePp0r4ONLBmxMF6EwDoAVpUIASPXgAC7bvyEiA9JcIBaTR226QRQBfNUM8Sj4JahR9RWZAk3QLK1sYhydeQWFiN3E2MGIvCIkFQyDMIJAVcAhZADE4AXwAFQBPSJQQZMIqmtlOgB4AWVaAYQgWPmgyYiw2YmIOAF5gAHJAXg3ARh3VoKw3AD4QIKA

## The current behavior

A parsing error occurs when an emoji is used within curly braces in the attribute value.

## The expected behavior

The emoji should be correctly parsed within curly braces without causing a parsing error.
